app/models/work.rb

```diff
@@ -0,0 +1,57 @@
+class Work < ApplicationRecord
+  has_many :votes
+  has_many :users, through: :votes
+
+  validates :category, presence: true
+  validates :category, inclusion: { in: ["movie", "book", "album"], message: "%{value} is not a valid category" }
+
+  validates :title, presence: true
+
+  validates :title, uniqueness: { scope: :category,
+    message: "The title should be unique within its category" }
+
+  validates :creator, presence: true
+
+  validates :publication_year, presence: true, numericality: { only_integer: true, less_than: (DateTime.now.year + 1), more_than: 0 },  length: { is: 4 }
+
+  def number_of_vote
+    vote_num = 0
+    votes.each do |vote|
+      vote_num += 1
+    end
+    return vote_num
+  end
+
+  def self.top_work
+    max_vote = 0
+    max_work = nil
+    Work.all.each do |work|
```

Comment:
This could be done with a where method.  

```ruby
  def self.top_ten(category)
    where(category: category).order(vote_count: :desc).limit(10)
  end
```